special.c

```diff
@@ -49,6 +49,19 @@ static request_t *special_unlink(object_t *o, request_t *r)
 }
 
 
+static request_t *special_devctl(object_t *o, request_t *r)
+{
+	unsigned long request;
+
+	ioctl_unpack(&r->msg, &request, NULL);
+	if (request == TCGETS) {
+		rq_setResponse(r, -ENOTTY);
+	}
```

Comment:
missing else, error code is random.
```suggestion
	int err = -EINVAL;

	ioctl_unpack(&r->msg, &request, NULL);
	if (request == TCGETS) {
		err = -ENOTTY; /* handle isatty() correctly */
	}
	rq_setResponse(r, err);

```